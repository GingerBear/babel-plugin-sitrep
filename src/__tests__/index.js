const pluginTester = require('babel-plugin-tester')
const plugin = require('../index')

pluginTester({
  plugin: plugin,
  pluginName: 'sitrep',
  snapshot: true,
  tests: [
    {
      title: 'should not add console logs',
      code: `
        function bar () {
          var a = 'foo'
          const b = 'bar'
          let c = [a, b].map(x => x)
          return c.join('-')
        }
        
        var cb = x => x.charAt(0)
        
        var cb = x => {
          x = x + 2
          x.charAt(0)
          return x
        }
        
        var a = function () {
          return 'foo'
        }
        
        const obj = {
          fn() {
            let a = 5
            return a + 5
          }
        }
        
        class Boom {
          fire() {
            let a = 2
            
            return a + 5
          }
        }
      `
    },
    {
      title: 'function',
      code: `
        // sitrep
        function bar () {
          var a = 'foo'
          const b = 'bar'
          let c = [a, b].map(x => x)
          return c.join('-')
        }
      `
    },
    {
      title: 'arrow function expression (shorthand arrow fn)',
      code: `
        // sitrep
        var cb = x => x.charAt(0)
      `
    },
    {
      title: 'arrow function assignment',
      code: `
        // sitrep
        var cb = x => {
          x = x + 2
          x.charAt(0)
          return x
        }
      `
    },
    {
      title: 'function assignment',
      code: `
        // sitrep
        var a = function () {
          return 'foo'
        }
      `
    },
    {
      title: 'object properties',
      code: `
        const obj = {
          // sitrep
          fn() {
            const { a, b, c = 'foo', d: alias } = x;
            return a + b + c + alias;
          }
        }
      `
    },
    {
      title: 'array properties',
      code: `
        // sitrep
        fn() {
          const [a, b, c, d] = x.split("_");
          return a + b + c + alias;
        }
      `
    },
    {
      title: 'class methods',
      code: `
        class Boom {
          // sitrep
          fire() {
            let a = 2
            
            return a + 5
          }
        }
      `
    },
    {
      title: 'no function parent',
      code: `
        // sitrep
        if (a) {
          let a = 2
          
          let b = a + 5
        }
      `
    },
    {
      title: 'arrow fn expression callback',
      code: `
        myFn(
          // sitrep
          (x) => x
        )
      `
    },
    {
      title: 'function declaration with multiple returns',
      code: `
        myFn(
          // sitrep
          function(err, vars) {
            if (err) {
              return err
            }
            
            vars = vars.map(x => x)
            return vars
          }
        )
      `
    },
    {
      title: 'collapsed option',
      code: `
        // sitrep
        function fn(a) {          
          a = a.map(x => x)
          return a
        }
      `,
      pluginOptions: {
        collapsed: false
      }
    },
    {
      title: 'custom label',
      code: `
        // ðŸ”¬
        function fn(a) {          
          a = a.map(x => x)
          return a
        }
      `,
      pluginOptions: {
        label: 'ðŸ”¬'
      }
    },
    {
      title: 'function has a for loop',
      code: `
        // sitrep
        function test() {
            var sum = 0;
            for (var i = 0; i < 5; i++) {
                sum = sum + i;
            }
            return sum;
        }
      `
    },
    {
      title: 'function has a for-of loop',
      code: `
        // sitrep
        function test(arr) {
            var sum = 0;
            for (const value of arr) {
                sum += value;
            }
        }
      `
    },
    {
      title: 'class property',
      code: `
        class Foo {
          // sitrep
          handleClick = (e) =>{
            const foo = e.target.innerText;
            // some actions based on foo
            e.preventDefault()
          }
        }
      `,
      babelOptions: {
        babelrc: true,
        filename: __filename
      }
    }
  ]
})
