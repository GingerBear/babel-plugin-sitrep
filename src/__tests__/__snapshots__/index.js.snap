// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`arrow fn expression callback 1`] = `
"
myFn(
  // sitrep
  (x) => x
)

      â†“ â†“ â†“ â†“ â†“ â†“

myFn(
// sitrep
function (x) {
  console.groupCollapsed(\\"function: (x) => x\\");
  var _returnValue = x;
  console.log(\\"Return Value:\\", _returnValue);
  console.groupEnd(\\"function: (x) => x\\");
  return _returnValue;
});
"
`;

exports[`arrow function assignment 1`] = `
"
// sitrep
var cb = x => {
  x = x + 2
  x.charAt(0)
  return x
}

      â†“ â†“ â†“ â†“ â†“ â†“

// sitrep
var cb = function (x) {
  console.groupCollapsed(\\"cb\\");

  x = x + 2;
  console.log(\\"x: \\", x);
  x.charAt(0);
  var _returnValue = x;
  console.log(\\"Return Value:\\", _returnValue);
  console.groupEnd(\\"cb\\");
  return _returnValue;
};
"
`;

exports[`arrow function expression (shorthand arrow fn) 1`] = `
"
// sitrep
var cb = x => x.charAt(0)

      â†“ â†“ â†“ â†“ â†“ â†“

// sitrep
var cb = function (x) {
  console.groupCollapsed(\\"cb\\");

  var _returnValue = x.charAt(0);

  console.log(\\"Return Value:\\", _returnValue);
  console.groupEnd(\\"cb\\");
  return _returnValue;
};
"
`;

exports[`class methods 1`] = `
"
class Boom {
  // sitrep
  fire() {
    let a = 2
    
    return a + 5
  }
}

      â†“ â†“ â†“ â†“ â†“ â†“

class Boom {
  // sitrep
  fire() {
    console.groupCollapsed(\\"fire\\");

    let a = 2;

    console.log(\\"a: \\", a);

    var _returnValue = a + 5;

    console.log(\\"Return Value:\\", _returnValue);
    console.groupEnd(\\"fire\\");
    return _returnValue;
  }
}
"
`;

exports[`collapsed option 1`] = `
"
// sitrep
function fn(a) {          
  a = a.map(x => x)
  return a
}

      â†“ â†“ â†“ â†“ â†“ â†“

// sitrep
function fn(a) {
  console.group(\\"fn\\");

  a = a.map(x => x);
  console.log(\\"a: \\", a);
  var _returnValue = a;
  console.log(\\"Return Value:\\", _returnValue);
  console.groupEnd(\\"fn\\");
  return _returnValue;
}
"
`;

exports[`custom label 1`] = `
"
// ðŸ”¬
function fn(a) {          
  a = a.map(x => x)
  return a
}

      â†“ â†“ â†“ â†“ â†“ â†“

// ðŸ”¬
function fn(a) {
  console.groupCollapsed(\\"fn\\");

  a = a.map(x => x);
  console.log(\\"a: \\", a);
  var _returnValue = a;
  console.log(\\"Return Value:\\", _returnValue);
  console.groupEnd(\\"fn\\");
  return _returnValue;
}
"
`;

exports[`function 1`] = `
"
// sitrep
function bar () {
  var a = 'foo'
  const b = 'bar'
  let c = [a, b].map(x => x)
  return c.join('-')
}

      â†“ â†“ â†“ â†“ â†“ â†“

// sitrep
function bar() {
  console.groupCollapsed('bar');

  var a = 'foo';
  console.log('a: ', a);
  const b = 'bar';
  console.log('b: ', b);
  let c = [a, b].map(x => x);
  console.log('c: ', c);

  var _returnValue = c.join('-');

  console.log('Return Value:', _returnValue);
  console.groupEnd('bar');
  return _returnValue;
}
"
`;

exports[`function assignment 1`] = `
"
// sitrep
var a = function () {
  return 'foo'
}

      â†“ â†“ â†“ â†“ â†“ â†“

// sitrep
var a = function () {
  console.groupCollapsed('a');
  var _returnValue = 'foo';
  console.log('Return Value:', _returnValue);
  console.groupEnd('a');

  return _returnValue;
};
"
`;

exports[`function declaration with multiple returns 1`] = `
"
myFn(
  // sitrep
  function(err, vars) {
    if (err) {
      return err
    }
    
    vars = vars.map(x => x)
    return vars
  }
)

      â†“ â†“ â†“ â†“ â†“ â†“

myFn(
// sitrep
function (err, vars) {
  console.groupCollapsed(\\"function: function(err, vars) {\\");

  if (err) {
    var _returnValue = err;
    console.log(\\"Return Value:\\", _returnValue);
    console.groupEnd(\\"function: function(err, vars) {\\");

    return _returnValue;
  }

  vars = vars.map(x => x);
  console.log(\\"vars: \\", vars);
  var _returnValue2 = vars;
  console.log(\\"Return Value:\\", _returnValue2);
  console.groupEnd(\\"function: function(err, vars) {\\");
  return _returnValue2;
});
"
`;

exports[`function has a for loop 1`] = `
"
// sitrep
function test() {
    var sum = 0;
    for (var i = 0; i < 5; i++) {
        sum = sum + i;
    }
    return sum;
}

      â†“ â†“ â†“ â†“ â†“ â†“

// sitrep
function test() {
    console.groupCollapsed(\\"test\\");

    var sum = 0;
    console.log(\\"sum: \\", sum);
    for (var i = 0; i < 5; i++) {
        sum = sum + i;
        console.log(\\"sum: \\", sum);
    }
    var _returnValue = sum;
    console.log(\\"Return Value:\\", _returnValue);
    console.groupEnd(\\"test\\");
    return _returnValue;
}
"
`;

exports[`function has a for-of loop 1`] = `
"
// sitrep
function test(arr) {
    var sum = 0;
    for (const value of arr) {
        sum += value;
    }
}

      â†“ â†“ â†“ â†“ â†“ â†“

// sitrep
function test(arr) {
    console.groupCollapsed(\\"test\\");

    var sum = 0;
    console.log(\\"sum: \\", sum);
    for (const value of arr) {
        sum += value;
        console.log(\\"sum: \\", sum);
    }
    console.groupEnd(\\"test\\");
}
"
`;

exports[`no function parent 1`] = `
"
// sitrep
if (a) {
  let a = 2
  
  let b = a + 5
}

      â†“ â†“ â†“ â†“ â†“ â†“

// sitrep
if (a) {
  let a = 2;

  let b = a + 5;
}
"
`;

exports[`object properties 1`] = `
"
const obj = {
  // sitrep
  fn() {
    const { a, b, c = 'foo', d: alias } = x;
    return a + b + c + alias;
  }
}

      â†“ â†“ â†“ â†“ â†“ â†“

const obj = {
  // sitrep
  fn() {
    console.groupCollapsed('fn');

    const { a, b, c = 'foo', d: alias } = x;
    console.log('a', a);
    console.log('b', b);
    console.log('c', c);
    console.log('alias', alias);

    var _returnValue = a + b + c + alias;

    console.log('Return Value:', _returnValue);
    console.groupEnd('fn');
    return _returnValue;
  }
};
"
`;

exports[`should not add console logs 1`] = `
"
function bar () {
  var a = 'foo'
  const b = 'bar'
  let c = [a, b].map(x => x)
  return c.join('-')
}

var cb = x => x.charAt(0)

var cb = x => {
  x = x + 2
  x.charAt(0)
  return x
}

var a = function () {
  return 'foo'
}

const obj = {
  fn() {
    let a = 5
    return a + 5
  }
}

class Boom {
  fire() {
    let a = 2
    
    return a + 5
  }
}

      â†“ â†“ â†“ â†“ â†“ â†“

function bar() {
  var a = 'foo';
  const b = 'bar';
  let c = [a, b].map(x => x);
  return c.join('-');
}

var cb = x => x.charAt(0);

var cb = x => {
  x = x + 2;
  x.charAt(0);
  return x;
};

var a = function () {
  return 'foo';
};

const obj = {
  fn() {
    let a = 5;
    return a + 5;
  }
};

class Boom {
  fire() {
    let a = 2;

    return a + 5;
  }
}
"
`;
